autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py

```diff
@@ -739,115 +743,72 @@ def load(cls, path, file_prefix="", reset_paths=False, verbose=True):
             If you want to load files with given prefix, can also pass arg: path = directory+file_prefix
         """
         path = path + file_prefix
-        obj = load_pkl.load(path = path + cls.model_file_name, verbose=verbose)
+        obj: TabularNeuralNetModel = load_pkl.load(path=path + cls.model_file_name, verbose=verbose)
         if reset_paths:
             obj.set_contexts(path)
-        obj.model = EmbedNet(architecture_desc=obj.architecture_desc, ctx=obj.ctx) # recreate network from architecture description
+        obj.model = EmbedNet(architecture_desc=obj._architecture_desc, ctx=obj.ctx)  # recreate network from architecture description
+        obj._architecture_desc = None
```

Comment:
This is to save space?
Are we sure this won't break repeated saving+reloading of the same model multiple times?

Reply:
ditto on previous comment